cogboard-webapp/src/components/UserLogin.js

```diff
@@ -1,31 +1,39 @@
 import React from 'react';
-import { useDispatch, useSelector } from 'react-redux';
+import {useDispatch, useSelector} from 'react-redux';
+import {Button, IconButton, TextField, Typography} from '@material-ui/core';
+import {AccountCircle, PowerSettingsNew} from '@material-ui/icons';
 
-import { useDialogToggle, useFormData } from '../hooks';
-import { login, logout } from '../actions/thunks';
-
-import { TextField, Button, IconButton, Typography } from '@material-ui/core';
-import { AccountCircle, PowerSettingsNew } from '@material-ui/icons';
-import { StyledFieldset } from './styled';
+import {useDialogToggle, useFormData, useSnackbarToggle} from '../hooks';
+import {login, logout} from '../actions/thunks';
+import {StyledFieldset} from './styled';
 import AppDialog from './AppDialog';
+import SnackbarWithVariant from "./SnackbarWithVariant";
 
 const UserLogin = () => {
   const dispatch = useDispatch();
-  const { values, handleChange } = useFormData({ username: '', password: ''});
+  const {values, handleChange} = useFormData({username: '', password: ''});
   const [dialogOpened, openDialog, handleDialogClose] = useDialogToggle();
   const errorMsg = useSelector(({app}) => app.loginErrorMessage);
   const jwToken = useSelector(({app}) => app.jwToken);
+  const [loginSnackbar, openLoginSnackbar, handleLoginSnackbarClose] = useSnackbarToggle();
+  const [logoutSnackbar, openLogoutSnackbar, handleLogoutSnackbarClose] = useSnackbarToggle();
+
+  const onLoginSuccess = () => {
+    handleDialogClose();
+    openLoginSnackbar();
+  };
 
   const handleLoginButtonClick = (credentials) => () => {
-    dispatch(login(credentials))
+    dispatch(login(credentials, onLoginSuccess))
```

Comment:
I think it would be better to set open prop based on Redux store state. Your solution works fine, but it is more imperative approach.

Reply:
Done. I used existence of jwt token to close the login dialog.